Shared/alerts/DynamicEntityLayerAlertTarget.cpp

```diff
@@ -0,0 +1,197 @@
+  m_entityGraphics(QMap<quint64, Graphic*>{})
+{
+  // subscribe to the entity received signal from the source of the dynamic layer
+  connect(m_dynamicEntityLayer->dataSource(), &DynamicEntityDataSource::dynamicEntityReceived, this, [this](DynamicEntityInfo *info)
+  {
+    // check the new entity for a connection and mark the info as delete later
+    auto dynamicEntity = info->dynamicEntity();
+    connectEntityGraphic(dynamicEntity);
+    info->deleteLater();
+  });
+
+  // subscribe to the observation received signal (this essentially serves as the 'update' signal for the dynamic layer type)
+  connect(m_dynamicEntityLayer->dataSource(), &DynamicEntityDataSource::dynamicEntityObservationReceived, this, [this](DynamicEntityObservationInfo *observationInfo)
+  {
+    // check the entity for a connection and mark the observation as delete later
+    auto dynamicEntity = observationInfo->observation()->dynamicEntity();
+    connectEntityGraphic(dynamicEntity);
+    observationInfo->deleteLater();
+  });
+
+  // subscribe to the purged signal to remove any graphics from the lookup
+  connect(m_dynamicEntityLayer->dataSource(), &DynamicEntityDataSource::dynamicEntityPurged, this, [this](DynamicEntityInfo *info)
+  {
+    // ensure the graphic is actually in the lookup
+    auto dynamicEntity = info->dynamicEntity();
+    if (m_entityGraphics.contains(dynamicEntity->entityId()))
+    {
+      // get a cleanup reference to the graphic that was stored and remove it from the lookup
+      std::unique_ptr<Graphic> graphic(m_entityGraphics[dynamicEntity->entityId()]);
+      m_entityGraphics.remove(dynamicEntity->entityId());
+
+      // signal the data has changed and rebuild the quad tree
+      rebuildQuadtree();
+      emit dataChanged();
+    }
+    info->deleteLater();
+  });
+}
+
+/*!
+  \brief Destructor.
+ */
+Dsa::DynamicEntityLayerAlertTarget::~DynamicEntityLayerAlertTarget()
+{
+}
+
+/*!
+  \brief Returns the list of \l Esri::ArcGISRuntime::Geometry which are in the \a targetArea.
+
+  \note No exact intersection tests are carried out to create this list.
+ */
+QList<Geometry> DynamicEntityLayerAlertTarget::targetGeometries(const Esri::ArcGISRuntime::Envelope &targetArea) const
+{
+  // if the quadtree has been built, use  it to return the set of candidate geometries
+  if (m_quadtree) { return m_quadtree->candidateIntersections(targetArea); }
+
+  // otherwise, return all of the geometry in the overlay
+  QList<Geometry> geometries;
+  for (auto dynamicEntity : m_entityGraphics)
+  {
+    if (!dynamicEntity) { continue; }
+    geometries.append(dynamicEntity->geometry());
+  }
+
+  return geometries;
+}
+
+/*!
+  \brief Returns an empty QVariant.
+ */
+QVariant DynamicEntityLayerAlertTarget::targetValue() const
+{
+  return QVariant{};
+}
+
+/*!
+  \internal
+
+  Connect signals etc. for a new \a dynamicEntity.
+ */
+void DynamicEntityLayerAlertTarget::connectEntityGraphic(Esri::ArcGISRuntime::DynamicEntity *dynamicEntity)
```

Comment:
Nitpick, but in the .cpp files we typically are `using namespace Esri::ArcGISRuntime`, so you don't need to fully qualify here.
```suggestion
void DynamicEntityLayerAlertTarget::connectEntityGraphic(DynamicEntity* dynamicEntity)
```

Reply:
Forgot to clean these up from the Refactor > AddDefinition to the class where the using namespace makes the full qualifier unnecessary.